### DISCUSSION TITLE
Upgrade k8s added comments back in machineconfig

### DISCUSSION THREAD

**Discussion by Hr46ph** [siderolabs/talos#9452] [Q&A]
Last night I ran `talosctl --nodes <controlplane node> upgrade-k8s --to 1.31.1` to upgrade from 1.30.

My machineconfigs are generated with Ansible for each node using templates and a patch file. During that process, it removes all the comments from the original file. The file that remains contains only the config parts that I need. This keeps the config clean and readable. This is pushed to the nodes. When using `talosctl edit mc`, its short and easy to read.

What happened last night is that after the kubernetes upgrade, the 2 worker nodes have all the comments added back to the machineconfig, while the 3 control planes remain without comments.

Besides the point that its just comments and doesn't do anything, I find it very weird that it did this and even weirder it would do this only on the workers and not on the control planes.

No harm done but its a bit of a hassle to clean them again and push the mc's back to the nodes. I wanted to report to make you aware of this as I believe this shouldn't have happened.

**Comment by smira** [comment]
`talosctl upgrade-k8s` patches the machine configuration on the node itself, if you want source of truth to be your Ansible automation, you shouldn't be using `talosctl upgrade-k8s`, but implement an equivalent of it using your own automation via Ansible.

You can also use `talosctl upgrade-k8s`, bump Kubernetes version on Ansible side, and re-apply the configs, if the only diff are the comments, it would be no-op.

If you are looking for a better way to manage machine configuration combined with upgrades, you can take a look at [Omni](https://github.com/siderolabs/omni).

**Reply by Hr46ph** [reply]
Sorry I wasn't very clear. I only use Ansible during deployment because it handles yaml very well, its easy with variables and I know Ansible very well. After the initial deployment of the cluster, I do not use Ansible to manage machine configs. I shouldn't have mentioned it as it only confuses and is unrelated to the issue at hand.

Either way, what led me to posting about this is that my machine configs are stripped of all comments. After `talosctl upgrade-k8s` on the 2 worker nodes the comments were added back while on the control planes they were not.

I would personally prefer it wouldn't add the comments back at all (why would it do that anyways?), but its also not consistent to add it back on one node and not on the other.

**Reply by smira** [reply]
It patches the machine configuration, so if there was a change, it will regenerate the machine config after patching, and will add comments back. You can configure it with `--with-docs=false --with-examples=false`.
